.ci/post_review.py

```diff
@@ -0,0 +1,214 @@
+                ":", maxsplit=4)
+            relative_path = os.path.relpath(abs_file_path, root_path)
+
+            k = 1
+            body = ""
+            while i+k < n and not ("error" in content[i+k] or "warning" in content[i+k]):
+                body += "\n" + content[i +
+                                       k].replace(abs_file_path, relative_path)
+                k += 1
+            i += k
+
+            message = message.strip().replace("'", "`")
+            body = body.strip()
+
+            comment_text = f"""{message}
+```cpp
+{body}
+```
+Reported as{severity} by clang-tidy.
+"""
+            parsed_report.append((relative_path, line_number, comment_text))
+    return parsed_report
+
+
+def file_filter_report(reports, diff):
+    changed_files = []
+    for file in diff:
+        changed_files.append(file.target_file[2:])
+
+    filtered_report = []
+    for report in reports:
+        rel_path, line_number, comment_text = report
+        if rel_path in changed_files:
+            filtered_report.append(report)
+
+    return filtered_report
+
+
+def line_filter_report(reports, diff):
+    files_in_report = [i[0] for i in reports]
+    filtered_report = []
+
+    lookup = {}
+    for file in diff:
+        if file.target_file[2:] in files_in_report:
+            for hunk in file:
+                for line in hunk.target_lines():
+                    if line.is_added:
+                        if file.target_file[2:] not in lookup:
+                            lookup[file.target_file[2:]] = list()
+                        lookup[file.target_file[2:]].append(
+                            int(line.target_line_no))
+
+    for report in reports:
+        fileName, number, _ = report
+        number = int(number)
+        if number in lookup[fileName]:
+            filtered_report.append(report)
+
+    return filtered_report
+
+
+def line_to_position(reports, diff):
+    lookup = {}
+    for file in diff:
+        filename = file.target_file[2:]
+        lookup[filename] = {}
+        pos = 1
+        for hunk in file:
+            for line in hunk:
+                if not line.is_removed:
+                    lookup[filename][line.target_line_no] = pos
+                pos += 1
+            pos += 1
+    new_reports = []
+    for report in reports:
+        x, y, z = report
+        new_reports.append((x, lookup[x][int(y)], z))
+
+    return new_reports
+
+
+def comment_lgtm(pr_handle):
+    lgtm = 'This Pull request Passed all of clang-tidy tests. :+1:'
+    comments = pr_handle.get_issue_comments()
+
+    for comment in comments:
+        if comment == lgtm:
+            print("Already posted LGTM!!")
+            return
+
+    pr_handle.create_issue_comment(lgtm)
+
+
+def post_review(reports, pr_handle):
+    comments = pr_handle.get_review_comments()
+    for comment in comments:
+        target = (comment.path, comment.position, comment.body)
+        if target in comments:
```

Comment:
```suggestion
        if target in reports:
```

Eureka! found it. Will continue the work next day.